plugins/Diagnostics/Diagnostic/DiagnosticResult.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+/**
+ * The result of a diagnostic.
```

Comment:
> Are there any conventions users of this class should know about? Ie, should users use HTML in their messages if they want

Good point. Ideally there shouldn't be any formatting here, unfortunately I did the refactoring with the "lowest effort possible", so the current situation is still not perfect on that point. So I'd rather not say "it's OK to use HTML" (because it's not) or say "it shouldn't be HTML" because many existing diagnostics use HTML so it would be confusing.

So the current status is: shhhh don't talk about it ;)